core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -1060,9 +1124,14 @@ class TimeSeriesShard(val dataset: Dataset,
             if (isActivelyIngesting(partitionObj.partID))
               logger.warn(s"Partition ${partitionObj.partID} is ingesting, but it was eligible for eviction. How?")
             if (endTime == PartKeyLuceneIndex.NOT_FOUND || endTime == Long.MaxValue) {
-              logger.warn(s"endTime ${endTime} was not correct. how?", new IllegalStateException())
+              logger.warn(s"endTime $endTime was not correct. how?", new IllegalStateException())
             } else {
               logger.debug(s"Evicting partId=${partitionObj.partID} from dataset=${dataset.ref} shard=$shardNum")
+              // add the evicted partKey to a bloom filter so that we are able to quickly
+              // find out if a partId has been assigned to an ingesting partKey before a more expensive lookup.
+              evictedPartKeys.add(PartKey(partitionObj.partKeyBase, partitionObj.partKeyOffset))
+              shardStats.evictedPkBloomFilterSize.set(evictedPartKeys.approximateElementCount())
```

Comment:
The approx element count of a bloom filter is like O(n) operation.  I suggest we don't set this every time, but on a minutely schedule (like all the other expensive stats we update at the end of flush cycle).

Reply:
But it would not be thread-safe to do a read on another thread while writes are happening to the bloom filter. Technically speaking, it is constant time since the size of the bloom filter is constant. But I get your reasoning. Let me think about other ways too...

Reply:
I am thinking it is acceptable since we set this only in the case where a partition is evicted. This should happen rarely.